### ISSUE TITLE
Bug with recipes not working / Items not craftable

### ISSUE THREAD

**Issue by Severdez** [NyaaCat/RPGitems-reloaded#274]
Sometimes the recipe of an item is not working. It needs to be "reset" with `/rpgitem ITEM recipe 1` to work again. I can't say what causes this bug, as it just happens randomly. We are not restarting the server or make any changes to the server. The bug occurs with just the server running normally.

Version: 1.11-v3.5.275

**Comment by Formula350** [comment]
Can you provide the recipe so others of us can attempt to re-create the problem? A common format I see used in forums to show a recipe is:
A, A, B
A, C, A
C, A, A
A = EMPTY
B = Diamond
C = Stick

**Comment by Severdez** [comment]
That is actually hard to say, because it happens with so many items. One example would be an upgraded armor set. You craft the first set like any other Vanilla armor. For the second you have to put any part of the first set in the middle and an upgrace material above. The upgrade material is crafted with 9x the item, that is used for the 1st tier armor. EVERY item needed in this process is a RPGItem that is given by playerkills or by commandblock over time.

The weird thing is, that the 1st tier armor could be crafted without problems, but NONE of the 2nd tier armor worked. Sometimes even the 1st tier armor is bugging.

**Comment by Severdez** [comment]
Oh I see. It makes a difference if the RPGItem is given by command or crafted. The item ID shown when doing the essentials command `/iteminfo` is different. Knowing that I think I can find a workaround for my case. But I can imagine that it still is annoying in many other scenarios.

**Comment by Formula350** [comment]
That's actually a thing I've been experiencing for years now, and yes it's *INCREDIBLY* annoying. Where it bites me in the ass is with the Quests plugin. Usually when we create a quest that requires an RPGItem (which is quite often) we'll just use the `/rpgitem <item> give` to have it on hand for input... Unfortunately, the issue ends up being that when a player comes to try and hand their item in, the Quests plugin is like "Sorry, this isn't the item the NPC is looking for..." which of course causes frustration to no end for staff *and* players. 

Curiously though, I had within the last week made a couple items and one of them just so happened to be an upgrade much like you're doing. For the upgrade recipe, it too utilized the crafted RPGItem in the center of the recipe; however, I've yet to hear anyone have a problem with crafting the upgrade, despite the first "crafted" item I used in the recipe being acquired via the give-command. I'll have to do some testing to see if I can see if that happens to me, as well.

I can indeed confirm that it's an issue with RPGItems, and has been forever. The problem, as far as I've been able to determine over the years, is just in the `Display` and `Hand/Type` entries, as the difference between rpgitems-given and crafted are a bunch of random colorcode padding that is added/removed from the metadata. In the case of the Hand/Type, it's at the beginning ***and*** end of that lore line.  
--Example taken from the Quests.yml file, which was 'added' to the plugin using the in-game quest creation system while holding the item in your hand *(bold colorcode are the plugin-added padding, the rest was added by me during creation)*:

> _- name-STONE_SWORD:amount-1:displayname-**§0§0§0§0§0§0§7§1§6§l**§eStar§r of the §4Nether:lore-**§0§0§0§0§0§0§7§1**§fTwo handed                       Claymore **§c§a§f§e§0§0§0§0**:lore-§f15-20 Damage:lore-§a50% chance to apply wither:lore-§aGrants regeneration 2 for 8.0s:lore-§e§o"The brightest point visible from within:lore-§e§othe Nether, forged into a mighty:lore-§e§osword.":lore-§7[Hit enemy to trigger Wither 2]:lore-§f§a[Right-click to activate Regeneration 2]:lore-§fCraftable §6Legendary Weapon_


To be honest, I don't even know *why* RPGItems even ***adds*** a bunch of extra color codes to the front of the item name... Unless it's used by RPGItems to recognize (authenticate) it as an `RPGItem` and not just some vanilla player-created item?
